### DISCUSSION TITLE
Using Model::hasRole('***') in the booted method not working as expected

### DISCUSSION THREAD

**Discussion by strotgen** [spatie/laravel-permission#1862] [Q&A] [answered]
Hi!

I am trying to use the _hasRole_ in the booted method of a model but it is not working as expected. It is checking against all the roles in the database, not just against the roles of the current auth user. 

```php
protected static function booted()
{
     if (auth()->check() && auth()->user()->hasRole('manager')) {
         static::addGlobalScope(...);
     }
}
```

This is happening if I use other methods. For example, using `auth()->user()->roles` returns all roles from all users in the database. This is the generated query:

```sql
select
  "roles".*,
  "model_has_roles"."model_id" as "pivot_model_id",
  "model_has_roles"."role_id" as "pivot_role_id",
  "model_has_roles"."model_type" as "pivot_model_type"
from
  "roles"
  inner join "model_has_roles" on "roles"."id" = "model_has_roles"."role_id"
```

**For some reason, it is failing to add the _where_ clause**:
```sql
where
  "model_has_roles"."model_id" = 2
  and "model_has_roles"."model_type" = 'App\Models\User'
```

`auth()->user()` is working fine. I even tried doing something like `User::first()->roles` inside the booted method but does not work either. 

Does anybody knows what is happening or how to fix this? I am guessing this has something to do with the way laravel executes, but not sure.

**Comment by erikn69** [comment]
package version ? laravel version? 
could you create a minimal Laravel app that demonstrates the issue? Or maybe create a failing tests for us to look at?

> I am guessing this has something to do with the way laravel executes

I suspect the same

**Comment by strotgen** [comment, marked as answer]
Hi @erikn69, thanks for your answer. 

I have doing some debugging and research and it seems that Laravel relationships are not available during the `booted` method, so that's why the `hasRole('manager')` fails to append the where clause. I am not 100 percent sure this is what is happening, but seems to be the case. For now, my solution was to do the query myself, not using the Laravel relations.

**Reply by erikn69** [reply]
>it seems that Laravel relationships are not available during the booted method

Thanks for the info, I didn't know that
